public/components/GeoPlot.vue

```diff
@@ -490,32 +516,41 @@ export default Vue.extend({
       });
     },
 
-    paint() {
-      if (!this.map) {
-        // NOTE: this component re-mounts on any change, so do everything in here
-        this.map = leaflet.map(this.mapID, {
-          center: [30, 0],
-          zoom: 2
-        });
-        if (this.mapZoom) {
-          this.map.setZoom(this.mapZoom, { animate: true });
-        }
-        if (this.mapCenter) {
-          this.map.panTo(
-            {
-              lat: this.mapCenter[1],
-              lng: this.mapCenter[0]
-            },
-            { animate: true }
-          );
-        }
-        this.baseLayer = leaflet.tileLayer(
-          "http://{s}.basemaps.cartocdn.com/light_all/{z}/{x}/{y}.png"
+    /**
+     * Create a Leaflet map, if it doesn't exist already, with basic defaults.
+     */
+    createMap() {
```

Comment:
the `paint()` was big, so I thought of moving out  the creation of the map because it was easy.